### ISSUE TITLE
Make babel-plugin-* and babel-preset-* dependencies direct dependencies (as opposed to devDependencies)

### ISSUE THREAD

**Issue by gajus** [prebid/Prebid.js#817]
## Type of issue

Bug

## Description

Make `babel-plugin-*` and `babel-preset-*` dependencies direct dependencies (as opposed to devDependencies). These are used by `babel-register` when resolving the dependencies.

> NOTE: This made me realize - you shouldn't include `es2015` preset in `.babelrc` thats being distributed in NPM package. Unless I am overlooking something (?), this will make webpack `babe-loader` pickup your config and negate the benefits of ES imports.

## Steps to reproduce

```bash
$ npm install prebid.js
$ cat <<'EOF' | > test.js
require('babel-register')({
  ignore: false;
}};
EOF
$ node --require ./test.js -eval 'import "prebid.js"'
```

The above will give an error:

> ReferenceError: Unknown plugin "transform-object-assign" specified in "/test/node_modules/prebid.js/.babelrc" at 0, attempted to resolve relative to "/test/node_modules/prebid.js"

### Expected results

Expect `babel-register` to be able to resolve `prebid.js` transpilers.

**Comment by mkendall07** [comment]
@gajus 
Can you tell us how you are consuming prebid using import? I'd like to understand the use case better. AFAIK, this won't work because there is no "distributable" with prebid published to NPM. It needs to be built after installing to run our build process.

**Comment by gajus** [comment]
> this won't work because there is no "distributable" with prebid published to NPM. It needs to be built after installing to run our build process.

We dropped Prebid.js from our stack, in part because of this reason.

I have spent the last month rewriting every adapter that we use (15+) and creating a modular header bidding library. We have made some unique improvements that allow us to perform header bidding considerably sooner than it is possible at all with Prebid.js (and making considerably less HTTP requests).

For now it will remain closed source. But I have raised a suggestion of making it open-source with the relevant people.

**Comment by mkendall07** [comment]
Thanks for the update. Even if you don't share the code, I  would be interested in the type of changes you have made for optimization in the adapters. Please share what you can. Thanks!

**Comment by gajus** [comment]
> I would be interested in the type of changes you have made for optimization in the adapters.

Just to be clear, we have not looked/ copied either the structure or the code of the adapters implementation in Prebid.js (except using it as a guide for Appnexus integration). Therefore, there is little in common.

There is not going to be much movement before the NY. Therefore, unlikely to share much before then. One thing that I can mention is that I have built a custom build system that inlines some of the SSP scripts into our codebase, which enables us to start bidding before loading their SDKs into the page. This alone has allowed us to start bidding around 300ms earlier than we were able before.
